src/python_testing/mdns_discovery/mdns_discovery.py

```diff
@@ -0,0 +1,275 @@
+            "type": service_info.type,
+            "server": service_info.server,
+            "port": service_info.port,
+            "addresses": service_info.parsed_addresses(),
+            "txt_record": service_info.decoded_properties,
+            "priority": service_info.priority,
+            "interface_index": service_info.interface_index,
+            "weight": service_info.weight,
+            "host_ttl": service_info.host_ttl,
+            "other_ttl": service_info.other_ttl
+        }
+        return service_info_dict
+
+    def _get_dut_operational_service_name(self) -> str:
+        """
+        Constructs the operational service name for the DUT.
+
+        This method generates a service name string based on the DUT's node ID and the compressed
+        fabric ID, following the operational service naming convention.
+
+        Returns:
+            str: A string representing the operational service name.
+        """
+        node_id = self._tc.dut_node_id
+        compressed_fabric_id = self._tc.default_controller.GetCompressedFabricId()
+        service_name = f'{compressed_fabric_id:016X}-{node_id:016X}.{MdnsType.OPERATIONAL.value}'
+        return service_name
+
+    def _getServiceInfo(self, mdns_type: MdnsType) -> List[Dict[str, any]]:
+        """
+        This method returns the discovered service information for a specified mDNS type.
+        It raises an exception if the discovery process has not been performed before calling this method.
+
+        Args:
+            mdns_type (MdnsType): The mDNS type for which to retrieve the service info.
+
+        Returns:
+            list: A list of discovered services of the specified mDNS type.
+
+        Raises:
+            DiscoveryNotPerformedError: If discovery has not been performed before calling this method.
+        """
+        if not self._discovery_performed:
+            raise DiscoveryNotPerformedError()
+
+        return self._discovered_services[mdns_type.name]
+
+    def getCommissionerServiceInfo(self) -> List[Dict[str, any]]:
+        """
+        Retrieves the service information for Commissioner services.
+
+        Returns:
+            list: A list of discovered Commissioner services.
+        """
+        return self._getServiceInfo(MdnsType.COMMISSIONER)
+
+    def getCommissionableServiceInfo(self) -> List[Dict[str, any]]:
+        """
+        Retrieves the service information for Commissionable services.
+
+        Returns:
+            list: A list of discovered Commissionable services.
+        """
+        return self._getServiceInfo(MdnsType.COMMISSIONABLE)
+
+    def getOperationalServiceInfo(self) -> List[Dict[str, any]]:
+        """
+        Retrieves the service information for Operational services.
+
+        Returns:
+            list: A list of discovered Operational services.
+        """
+        return self._getServiceInfo(MdnsType.OPERATIONAL)
+
+    def getBorderRouterServiceInfo(self) -> List[Dict[str, any]]:
```

Comment:
Please follow pep8 snake case naming conventions or existing conventions in Python as much as possible